test/e2e/ocpbugs48050_test.go

```diff
@@ -0,0 +1,611 @@
+				}
+				return nil
+			}); err != nil {
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			t.Fatalf("/healthz checks failed for route %s: %v", host, err)
+		}
+
+		// Once /healthz passes, proceed with probing
+		// /single-te and /duplicate-te endpoints.
+		for _, tc := range []struct {
+			path            string
+			responseChecker func(*http.Response, error) error
+		}{
+			{"/single-te", singleTransferEncodingResponseCheck},
+			{"/duplicate-te", duplicateTransferEncodingResponseCheck},
+		} {
+			url := fmt.Sprintf("http://%s%s", host, tc.path)
+			for j := 0; j < routeIndex(route.Name); j++ {
+				if err := httpGet(url, tc.responseChecker); err != nil {
+					t.Fatalf("GET request to %s failed: %v", url, err)
+				}
+			}
+		}
+	}
+
+	promClient, err := newPrometheusClient()
+	if err != nil {
+		t.Fatalf("failed to create prometheus client: %v", err)
+	}
+
+	// Wait for at least two metric scrapes from each router pod.
+	// While one scrape is sometimes sufficient, relying on just
+	// one scrape can make the overall test flaky. Two scrapes
+	// ensure more consistent and reliable test results.
+	newScrapes := 2
+	maxScrapeWaitDuration := 5 * time.Minute
+	t.Logf("Waiting up to %.0f minutes for router pod metric scrape counts to increment by %d", maxScrapeWaitDuration.Minutes(), newScrapes)
+	if err := waitForDefaultRouterScrapesIncrement(promClient, newScrapes, maxScrapeWaitDuration, func(initialCounts, currentCounts podScrapeCounts, allIncremented bool) {
+		for podName, initialCount := range initialCounts {
+			currentCount := currentCounts[podName]
+			targetCount := initialCount + float64(newScrapes)
+			status := "in progress"
+			if currentCount >= targetCount {
+				status = "complete"
+			}
+			t.Logf("Scrape count progress for pod %s: initial: %.0f, current: %.0f, target: %.0f (%s)",
+				podName, initialCount, currentCount, targetCount, status)
+		}
+	}); err != nil {
+		t.Fatalf("Error waiting for scrape counts to increment: %v", err)
+	}
+
+	duplicateTECounts, err := queryPrometheusForDuplicateTEHeaders(promClient, ns.Name, time.Minute)
+	if err != nil {
+		t.Fatalf("Failed to query Prometheus: %v", err)
+	}
+
+	t.Log("Detailed results:")
+	t.Logf("%-20s %-15s %-15s %-15s %-15s", "Route", "Termination", "Actual Count", "Expected Count", "Match?")
+
+	for _, route := range routeList.Items {
+		actualCount, exists := duplicateTECounts[route.Name]
+		routeIndex := routeIndex(route.Name)
+
+		var expectedCount float64
+
+		if route.Spec.TLS.Termination != routev1.TLSTerminationPassthrough {
+			expectedCount = float64(routeIndex)
+		}
+
+		success := actualCount == expectedCount
+
+		t.Logf("%-20s %-15s %-15.0f %-15.0f %-15v", route.Name, route.Spec.TLS.Termination, actualCount, expectedCount, success)
+
+		if !exists {
+			t.Errorf("Route %s not found in results", route.Name)
+		} else if !success {
+			t.Errorf("Mismatch for route %s: expected %.0f, got %.0f", route.Name, expectedCount, actualCount)
```

Comment:
Similar question, should we log the on the line 603 if we error with this message (error will be logged too)?

Reply:
Is there any harm in logging the state, followed by an error message if any. In a successful run you'll just see the state logged. If there's an error you get to keep both pieces.